src/tabpane/TabPane.ts

```diff
@@ -0,0 +1,181 @@
+	activeIndex?: number;
+	alignButtons?: Align;
+	loadingIndex?: number;
+	tabs?: TabConfig[];
+	onRequestTabChange?(index: number): void;
+	onRequestTabClose?(tabs: TabConfig[]): void;
+};
+
+export const TabPaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TabPane extends TabPaneBase<TabPaneProperties> {
+	private _loading: boolean;
+
+	private _renderTabButtons() {
+		const {
+			activeIndex = 0,
+			loadingIndex,
+			tabs = []
+		} = this.properties;
+
+		return tabs.map((tab, i) => w(TabButton, {
+			key: String(i),
+			active: i === (this._loading ? loadingIndex : activeIndex),
+			closeable: tab.closeable,
+			disabled: tab.disabled,
+			index: i,
+			loading: i === loadingIndex,
+			onClick: this.onTabClick,
+			onCloseClick: this.onCloseClick
+		}, [ tab.label || null ]));
+	}
+
+	private _renderTabs() {
+		const {
+			activeIndex = 0,
+			loadingIndex,
+			tabs = []
+		} = this.properties;
+
+		return tabs.filter((tab, i) => {
+			return i === (this._loading ? loadingIndex : activeIndex);
+		}).map((tab, i) => w(Tab, {
+			active: true,
+			loading: this._loading
+		}, [ this._loading ? 'Loading...' : (tab.content || null) ]));
+	}
+
+	protected onTabClick(index: number) {
+		const { onRequestTabChange } = this.properties;
+		onRequestTabChange && onRequestTabChange(index);
+	}
+
+	protected onCloseClick(index: number) {
+		let {
+			tabs = [],
+			onRequestTabClose
+		} = this.properties;
+
+		const newTabs = [...tabs];
+		newTabs.splice(index, 1);
+
+		onRequestTabClose && onRequestTabClose(newTabs);
+	}
+
+	protected updateActiveIndex() {
+		let {
+			activeIndex = 0,
+			tabs = [],
+			onRequestTabChange
+		} = this.properties;
+
+		if (tabs[activeIndex] && !tabs[activeIndex].disabled) {
+			return;
+		}
+
+		for (let i = 0; i < tabs.length; i++) {
+			if (!tabs[i].disabled) {
+				onRequestTabChange && onRequestTabChange(i);
+				break;
+			}
+		}
+	}
+
+	@onPropertiesChanged
+	protected onPropertiesChanged(evt: PropertiesChangeEvent<this, TabPaneProperties>) {
+		if (includes(evt.changedPropertyKeys, 'tabs') || includes(evt.changedPropertyKeys, 'activeIndex')) {
+			this.updateActiveIndex();
+		}
+		if (includes(evt.changedPropertyKeys, 'loadingIndex')) {
+			const index = evt.properties.loadingIndex;
+			this._loading = typeof index === 'number';
+		}
+	}
+
+	render(): DNode {
+		const { alignButtons } = this.properties;
+
+		let alignClass;
+		const children = [
+			v('div', {
+				key: 'buttons',
+				classes: this.classes(css.tabButtons)
+			}, this._renderTabButtons()),
+			v('div', {
+				key: 'tabs',
+				classes: this.classes(css.tabs)
+			}, this._renderTabs())
+		];
+
+		switch (alignButtons) {
+			case Align.right:
+				alignClass = css.alignRight;
+				children.reverse();
+				break;
+			case Align.bottom:
+				alignClass = css.alignBottom;
+				children.reverse();
+				break;
+			case Align.left:
+				alignClass = css.alignLeft;
+				break;
+		}
+
+		return v('div', {
```

Comment:
This one needs `role: tablist`, and if alignButtons is left or right, it'll need `aria-orientation` set to vertical (`horizontal` is the default, but it wouldn't hurt to set it explicitly either)